micronaut-camunda-bpm-feature/src/main/java/info/novatec/micronaut/camunda/bpm/feature/MnAdminUserConfiguration.java

```diff
@@ -0,0 +1,108 @@
+import org.camunda.bpm.engine.impl.persistence.entity.AuthorizationEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Singleton;
+
+import static org.camunda.bpm.engine.authorization.Authorization.ANY;
+import static org.camunda.bpm.engine.authorization.Authorization.AUTH_TYPE_GRANT;
+import static org.camunda.bpm.engine.authorization.Groups.CAMUNDA_ADMIN;
+import static org.camunda.bpm.engine.authorization.Permissions.ALL;
+
+/**
+ * Bean creating Camunda Admin User, Group and Authorizations if {@code camunda.bpm.admin-user.id} Property is present.
+ *
+ * @author Titus Meyer
+ */
+@Singleton
+@Requires(property = "camunda.bpm.admin-user.id")
+public class MnAdminUserConfiguration implements BeanCreatedEventListener<ProcessEngine> {
+    private static final Logger log = LoggerFactory.getLogger(MnAdminUserConfiguration.class);
+
+    private final Configuration.AdminUser adminUser;
+
+    public MnAdminUserConfiguration(Configuration configuration) {
+        adminUser = configuration.getAdminUser();
+    }
+
+    @Override
+    public ProcessEngine onCreated(BeanCreatedEvent<ProcessEngine> event) {
+        ProcessEngine processEngine = event.getBean();
+
+        final IdentityService identityService = processEngine.getIdentityService();
+        final AuthorizationService authorizationService = processEngine.getAuthorizationService();
+
+        if (!userAlreadyExists(identityService, adminUser)) {
+            createUser(identityService, adminUser);
+
+            // create group
+            if (identityService.createGroupQuery().groupId(CAMUNDA_ADMIN).count() == 0) {
+                Group camundaAdminGroup = identityService.newGroup(CAMUNDA_ADMIN);
+                camundaAdminGroup.setName("camunda BPM Administrators");
+                camundaAdminGroup.setType(Groups.GROUP_TYPE_SYSTEM);
+                identityService.saveGroup(camundaAdminGroup);
+            }
+
+            // create ADMIN authorizations on all built-in resources
+            for (Resource resource : Resources.values()) {
+                if (authorizationService.createAuthorizationQuery().groupIdIn(CAMUNDA_ADMIN).resourceType(resource).resourceId(ANY).count() == 0) {
+                    AuthorizationEntity userAdminAuth = new AuthorizationEntity(AUTH_TYPE_GRANT);
+                    userAdminAuth.setGroupId(CAMUNDA_ADMIN);
+                    userAdminAuth.setResource(resource);
+                    userAdminAuth.setResourceId(ANY);
+                    userAdminAuth.addPermission(ALL);
+                    authorizationService.saveAuthorization(userAdminAuth);
+                }
+            }
+
+            identityService.createMembership(adminUser.getId(), CAMUNDA_ADMIN);
+            log.info("creating initial Admin User: {}", adminUser.getId());
+        }
+
+        return processEngine;
+    }
+
+    static boolean userAlreadyExists(IdentityService identityService, Configuration.AdminUser adminUser) {
```

Comment:
* The signature would be cleaner with the "id" as a parameter.
* all "package private" methods should probably be declared as "protected" to allow overriding if needed.